spring-data-cassandra/src/main/java/org/springframework/data/cassandra/config/CassandraSessionFactoryBean.java

```diff
@@ -13,109 +13,153 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package org.springframework.data.cassandra.config;
 
+import static org.springframework.cassandra.core.cql.CqlIdentifier.cqlId;
+
 import java.util.Collection;
 
 import org.springframework.cassandra.config.CassandraCqlSessionFactoryBean;
 import org.springframework.data.cassandra.convert.CassandraConverter;
+import org.springframework.data.cassandra.core.CassandraAdminOperations;
 import org.springframework.data.cassandra.core.CassandraAdminTemplate;
 import org.springframework.data.cassandra.mapping.CassandraMappingContext;
 import org.springframework.data.cassandra.mapping.CassandraPersistentEntity;
 import org.springframework.util.Assert;
 
 import com.datastax.driver.core.KeyspaceMetadata;
 import com.datastax.driver.core.Metadata;
+import com.datastax.driver.core.Session;
 import com.datastax.driver.core.TableMetadata;
 
-import static org.springframework.cassandra.core.cql.CqlIdentifier.cqlId;
-
+/**
+ * Factory to create and configure a Cassandra {@link com.datastax.driver.core.Session} with support
+ * for executing CQL and initializing the database schema (a.k.a. keyspace).
+ *
+ * @author Mathew Adams
+ * @author David Webb
+ * @author John Blum
+ * @see com.datastax.driver.core.KeyspaceMetadata
+ * @see com.datastax.driver.core.TableMetadata
+ */
 public class CassandraSessionFactoryBean extends CassandraCqlSessionFactoryBean {
 
-	protected SchemaAction schemaAction = SchemaAction.NONE;
-	protected CassandraAdminTemplate admin;
-	protected CassandraConverter converter;
-	protected CassandraMappingContext mappingContext;
+	protected static final boolean DEFAULT_CREATE_IF_NOT_EXISTS = false;
+	protected static final boolean DEFAULT_DROP_TABLES = false;
+	protected static final boolean DEFAULT_DROP_UNUSED_TABLES = false;
+
+	private CassandraAdminOperations admin;
+
+	private CassandraConverter converter;
+
+	private SchemaAction schemaAction = SchemaAction.NONE;
 
 	@Override
 	public void afterPropertiesSet() throws Exception {
-
 		super.afterPropertiesSet();
 
-		Assert.notNull(converter);
+		Assert.state(converter != null, "Converter must not be null");
```

Comment:
I guess there's no consistent rule for exceptions in `afterPropertiesSet` - even the core framework throws `IllegalArgument`/`NullPointer`/other exceptions. Was just wondering, all good here.